### PR TITLE
feat(bazel): update to the latest `@microsoft/api-extractor`

### PR DESCRIPTION
**Pull request by alan-agius4**
Update usages of `@microsoft/api-extractor`  to the latest version

### CODE REVIEW

**Comment by alan-agius4** [comment]
Will need to investigate why certain declarations are being wrapped with double quotes examples;
```
declare class "ɵNgNoValidate" {
}
export { "ɵNgNoValidate" as ɵNgNoValidate }
export { "ɵNgNoValidate" as ɵangular_packages_forms_forms_z }

```

**Comment by alan-agius4** [comment]
Opened https://github.com/microsoft/web-build-tools/issues/1471

**Review by alexeagle — APPROVED** [review]
I assume the new version produces identical output, and that's why this PR doesn't include any test or golden file changes?

File: packages/bazel/src/api-extractor/index.ts [hunk 1]
```diff
@@ -55,51 +55,46 @@ export function runMain(
...
-    project: {
-      entryPointSourceFile: path.resolve(entryPoint),
     },
-    apiReviewFile: {
+    projectFolder: path.resolve(path.dirname(tsConfig)),
+    mainEntryPointFilePath: path.resolve(entryPoint),
+    apiReport: {
       enabled: !!apiReviewFolder,
-      apiReviewFolder: apiReviewFolder && path.resolve(apiReviewFolder),
+      // todo(alan-agius4): remove this folder name when the below issue is solved upstream
```
[the remark this answers is not in the document]
Comment (alan-agius4) [reply packages/bazel/src/api-extractor/index.ts]: Will to, though I am still blocked on this, due the 2 bugs at upstream.

**Comment by alan-agius4** [comment]
@alexeagle, the output is indeed the same.

**Pull request closed without merging**